### PR TITLE
[sovereign] FIX: Save metrics panic

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- 
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by axenteoctavian — DISMISSED** [review]

File: process/block/sovereignChainBlock.go [hunk 1]
```diff
@@ -792,6 +788,20 @@ func (scbp *sovereignChainBlockProcessor) ProcessBlock(headerHandler data.Header
...
 	}
 
+	newBody := body
+	defer func() {
+		if err != nil {
+			return
+		}
+
+		if !check.IfNil(shardHeader) {
+			go getMetricsFromHeader(headerHandler.ShallowClone(), uint64(txCounts.GetTotal()), scbp.marshalizer, scbp.appStatusHandler)
```
Comment (sasurobert) [inline process/block/sovereignChainBlock.go:798]: shardHeader in the go func ?

how can you have nil shardheader at this point ?

Even the body should not be nil.

Comment (mariusmihaic) [reply process/block/sovereignChainBlock.go]: Right, we can get rid of those.
Fixed as suggested

**Review by axenteoctavian — APPROVED** [review]

File: process/block/sovereignChainBlock.go [hunk 2]
```diff
@@ -792,6 +788,16 @@ func (scbp *sovereignChainBlockProcessor) ProcessBlock(headerHandler data.Header
 		return nil, nil, err
 	}
 
+	newBody := body
+	defer func() {
+		if err != nil {
```
Comment (sasurobert) [inline process/block/sovereignChainBlock.go:793]: this if does not have a reason to be here.

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: process/block/sovereignChainBlock.go (hunk 2 above)
Comment (mariusmihaic) [reply process/block/sovereignChainBlock.go:793]: You are absolutely right here, deleted it.

**Review by axenteoctavian — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**